feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radius_proxy.c

```diff
@@ -140,21 +140,31 @@ static bool radsec_download_cert(char *cert_name, char *dir_name, char *cert_url
 {
 	CURL *curl;
 	FILE *fp;
-	CURLcode res;
+	CURLcode curl_ret;
 	char path[200];
+	char dir_path[200];
 	char name[32];
 	char dir[32];
 	char *gw_clientcert = "/usr/opensync/certs/client.pem";
 	char *gw_clientkey = "/usr/opensync/certs/client_dec.key";
+	struct stat stat_buf;
 
 	strcpy(name, cert_name);
 	strcpy(dir, dir_name);
+	sprintf(dir_path, "/tmp/radsec/certs/%s", dir);
 	sprintf(path, "/tmp/radsec/certs/%s/%s", dir, name);
 
+	if (stat(dir_path, &stat_buf) == -1)
+	{
+		char cmd[200];
+		sprintf(cmd, "mkdir -p %s", dir_path);
+		system(cmd);
+	}
+
```

Comment:
From the captive portal code:
	struct stat st = {0};
	if (stat(path, &st) == -1)
		mkdir(path, 0755);

Reply:
mkdir() doesn't work for making nested directories in one shot.